### PR TITLE
Add get, getAll, delete, has and set support

### PR DESCRIPTION
**Pull request by yordis**
Related to #403 #124

### CODE REVIEW

File: test/fixture/cert.pem [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
-MIIBnDCCAUYCCQCgIldFYidw0zANBgkqhkiG9w0BAQUFADBVMQswCQYDVQQGEwJV
-UzELMAkGA1UECBMCQ0ExEjAQBgNVBAcTCVBhbG8gQWx0bzERMA8GA1UEChMIRm9y
-bURhdGExEjAQBgNVBAMTCWxvY2FsaG9zdDAeFw0xNTA2MTMxMzA3MTBaFw00MjEw
-MjgxMzA3MTBaMFUxCzAJBgNVBAYTAlVTMQswCQYDVQQIEwJDQTESMBAGA1UEBxMJ
-UGFsbyBBbHRvMREwDwYDVQQKEwhGb3JtRGF0YTESMBAGA1UEAxMJbG9jYWxob3N0
-MFwwDQYJKoZIhvcNAQEBBQADSwAwSAJBAOkE8JJzLQbVEo/vRVRqiwNEbiuEooJA
-099h1IJLtsPJVphSDBTtE0FLB0YIzT7bG44Drm/975XEZjoQ9TmYbTMCAwEAATAN
-BgkqhkiG9w0BAQUFAANBALSDmvKXb4cE622OdSRrDsnr8xUosetsAGSlGauWBKgc
-DNQhP6wSQHTAJT8wiVgSPYwnhcnzkZlwzdfMHMtzFIc=
+MIICpDCCAYwCCQDB+amVj19LbTANBgkqhkiG9w0BAQsFADAUMRIwEAYDVQQDDAls
```
Comment (yordis) [inline test/fixture/cert.pem]: I had to update the certs for be able to run in newest version of NodeJS

File: lib/form_data.js [hunk 2]
```diff
@@ -43,7 +87,39 @@ function FormData(options) {
...
+    return this.append(field, value, options);
+  }
+
+  // 3. If there are any entries in the context object’s entry list whose name
+  // is name, then replace the first such entry with entry and remove the
+  // others.
+
+  // Replace the firt entry
+  var newEntry = createEntry(field, value);
+  this._entryList[foundEntryIndex] = newEntry;
```
Comment (yordis) [inline lib/form_data.js]: Before doing this I need some help understanding why we need to do so many things in the `append` function.

File: lib/form_data.js [hunk 3]
```diff
@@ -9,12 +9,55 @@ var mime = require('mime-types');
...
 util.inherits(FormData, CombinedStream);
 
+/**
+ * @param {string} name the entry name
+ * @param {string} value the entry value
+ * @returns {Object}
+ */
+function createEntry(name, value/*, filename*/) {
+  /*
+    TODO: handle Blob and File
```
Comment (yordis) [inline lib/form_data.js]: Need help with this

**Comment by yordis** [comment]
@alexindigo hey mate could you give me a hand with this PR and also try to finalize it?

**Comment by alexindigo** [comment]
@yordis This PR looks ambitious and I do appreciate the end goal. Do you think we can add this functionality in smaller chunks, in more iterative way, rather than #AllTheThings at once? And it will be easier to land as well.

**Comment by softwareCobbler** [comment]
What happened to `FormData.set`? I'd like to cast a vote for it being included.

**Comment by nitreojs** [comment]
What about `.get()` and `.getAll()`? Seems like cool methods that I would use

**Pull request closed without merging**